### PR TITLE
HPCC-20168 Include Spark conditionally into platform package

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Change unpacked spark name to spark-hadoop

Changes to the packaging

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@xwang2713 Merged the changes I made to your original work and separated the installation PR from the configuration/running PR.

File: CMakeLists.txt [hunk 1]
```diff
@@ -197,6 +197,10 @@ if(APPLE)
     HPCC_ADD_SUBDIRECTORY(package)
 endif(APPLE)
 
+if(USE_SPARK)
```
Comment (rpastrana) [inline CMakeLists.txt]: this might be standard cmake naming convention, but this flag could be more self documenting as INTEGRATE_SPARK ??

**Review by rpastrana — APPROVED** [review]
@Michael-Gardner one minor comment, otherwise it's fine (I believe this code had already been reviewed)

File: CMakeLists.txt (hunk 1 above)
Comment (Michael-Gardner) [reply CMakeLists.txt]: USE_SPARK is the convention we use for including an optional feature in a package.  You can find examples of this in the commonSetup.cmake file, (ex. USE_PYTHON3).

**Comment by jpmcmu** [comment]
@Michael-Gardner looks good

**Review by xwang2713 — COMMENTED** [review]
Looks good

**Comment by Michael-Gardner** [comment]
@ghalliday ready for merge

**Comment by rpastrana** [comment]
@xwang2713 can we create an LN+spark build on Jenkins?

**Comment by xwang2713** [comment]
Already did: http://10.240.32.243/view/HPCC-7.0.0-rc1/job/LN-Candidate-with-Plugins-Spark-7.0.0-rc1/

**Comment by rpastrana** [comment]
Thank you @xwang2713

**Pull request merged**